### PR TITLE
Add accessible `<FileTree>` component to replace ASCII art

### LINKED ISSUE #2167: [A11Y] File lists are currently inaccessible ASCII art representations
Currently, when we're representing a file tree of some kind, we display something like the following:

An ASCII representation of a file tree, based on the output of the `tree` command

This doesn't read out well with accessible technologies such as screen-readers. Instead, we might want to consider building out an interactive component that supports assistive technologies better, as well as collapseable states.

An example of what a component might look like could be found in [Ant Design](https://ant.design/components/tree#components-tree-demo-directory) or [Fluent UI](https://developer.microsoft.com/en-us/fluentui#/controls/web/groupedlist)

# List of Impacted Files

- [x] [/old-translations/de/core-concepts/project-structure.md](https://github.com/withastro/docs/tree/main/old-translations/de/core-concepts/project-structure.md)
- [x] [/old-translations/de/guides/styling.md](https://github.com/withastro/docs/tree/main/old-translations/de/guides/styling.md)
- [x] [/old-translations/es/core-concepts/project-structure.md](https://github.com/withastro/docs/tree/main/old-translations/es/core-concepts/project-structure.md)
- [ ] [/src/pages/de/core-concepts/project-structure.md](https://github.com/withastro/docs/tree/main/src/pages/de/core-concepts/project-structure.md)
- [ ] [/src/pages/de/core-concepts/routing.md](https://github.com/withastro/docs/tree/main/src/pages/de/core-concepts/routing.md)
- [ ] [/src/pages/de/install/manual.md](https://github.com/withastro/docs/tree/main/src/pages/de/install/manual.md)
- [x] [/src/pages/en/core-concepts/project-structure.md](https://github.com/withastro/docs/tree/main/src/pages/en/core-concepts/project-structure.md)
- [x] [/src/pages/en/core-concepts/routing.md](https://github.com/withastro/docs/tree/main/src/pages/en/core-concepts/routing.md)
- [x] [/src/pages/en/guides/cms/contentful.md](https://github.com/withastro/docs/tree/main/src/pages/en/guides/cms/contentful.md)
- [x] [/src/pages/en/guides/cms/storyblok.md](https://github.com/withastro/docs/tree/main/src/pages/en/guides/cms/storyblok.md)
- [x] [/src/pages/en/install/manual.md](https://github.com/withastro/docs/tree/main/src/pages/en/install/manual.md)
- [x] [/src/pages/en/reference/publish-to-npm.md](https://github.com/withastro/docs/tree/main/src/pages/en/reference/publish-to-npm.md)
- [ ] [/src/pages/es/core-concepts/project-structure.md](https://github.com/withastro/docs/tree/main/src/pages/es/core-concepts/project-structure.md)
- [ ] [/src/pages/es/core-concepts/routing.md](https://github.com/withastro/docs/tree/main/src/pages/es/core-concepts/routing.md)
- [ ] [/src/pages/es/install/manual.md](https://github.com/withastro/docs/tree/main/src/pages/es/install/manual.md)
- [ ] [/src/pages/es/reference/publish-to-npm.md](https://github.com/withastro/docs/tree/main/src/pages/es/reference/publish-to-npm.md)
- [ ] [/src/pages/fr/core-concepts/project-structure.md](https://github.com/withastro/docs/tree/main/src/pages/fr/core-concepts/project-structure.md)
- [ ] [/src/pages/fr/core-concepts/routing.md](https://github.com/withastro/docs/tree/main/src/pages/fr/core-concepts/routing.md)
- [ ] [/src/pages/fr/install/manual.md](https://github.com/withastro/docs/tree/main/src/pages/fr/install/manual.md)
- [ ] [/src/pages/fr/reference/publish-to-npm.md](https://github.com/withastro/docs/tree/main/src/pages/fr/reference/publish-to-npm.md)
- [ ] [/src/pages/ja/core-concepts/project-structure.md](https://github.com/withastro/docs/tree/main/src/pages/ja/core-concepts/project-structure.md)
- [ ] [/src/pages/ja/core-concepts/routing.md](https://github.com/withastro/docs/tree/main/src/pages/ja/core-concepts/routing.md)
- [ ] [/src/pages/ja/install/manual.md](https://github.com/withastro/docs/tree/main/src/pages/ja/install/manual.md)
- [ ] [/src/pages/pt-br/core-concepts/project-structure.md](https://github.com/withastro/docs/tree/main/src/pages/pt-br/core-concepts/project-structure.md)
- [ ] [/src/pages/pt-br/core-concepts/routing.md](https://github.com/withastro/docs/tree/main/src/pages/pt-br/core-concepts/routing.md)
- [ ] [/src/pages/pt-br/install/manual.md](https://github.com/withastro/docs/tree/main/src/pages/pt-br/install/manual.md)
- [ ] [/src/pages/pt-br/reference/publish-to-npm.md](https://github.com/withastro/docs/tree/main/src/pages/pt-br/reference/publish-to-npm.md)
- [ ] [/src/pages/ru/core-concepts/project-structure.md](https://github.com/withastro/docs/tree/main/src/pages/ru/core-concepts/project-structure.md)
- [ ] [/src/pages/zh-cn/core-concepts/project-structure.md](https://github.com/withastro/docs/tree/main/src/pages/zh-cn/core-concepts/project-structure.md)
- [ ] [/src/pages/zh-cn/core-concepts/routing.md](https://github.com/withastro/docs/tree/main/src/pages/zh-cn/core-concepts/routing.md)
- [ ] [/src/pages/zh-cn/install/manual.md](https://github.com/withastro/docs/tree/main/src/pages/zh-cn/install/manual.md)
- [ ] [/src/pages/zh-cn/reference/publish-to-npm.md](https://github.com/withastro/docs/tree/main/src/pages/zh-cn/reference/publish-to-npm.md)
- [ ] [/src/pages/zh-tw/core-concepts/project-structure.md](https://github.com/withastro/docs/tree/main/src/pages/zh-tw/core-concepts/project-structure.md)
- [ ] [/src/pages/zh-tw/install/manual.md](https://github.com/withastro/docs/tree/main/src/pages/zh-tw/install/manual.md)

### PR DESCRIPTION
**Pull request by delucis**
## Summary

- Closes #2167
- Adds a `<FileTree>` component to render file hierarchies in a more accessible manner than the ASCII art we use currently. Also in a _fancier_ manner:
  - It’s collapsible!
  - It has file-type icons!
  - Zero JS!

## Authoring

Once you import the `FileTree` component into your file, authoring looks like this:

```md
<FileTree>
- src/
  - pages/
    - **index.astro**
- astro.config.mjs This is the Astro config file
- package.json
</FileTree>
```

1. A file or directory can be marked as highlighted by bolding it: `**index.astro**`.
2. Any text following the first space in an item is treated as a comment and styled accordingly, see the `astro.config` entry above.

## Screenshots

Here’s a before/after from the manual install guide:

| Before | After |
|--------|-------|
| image | image |

## How does this work

The `<FileTree>` component expects to be passed an unordered list. It passes this list through a rehype pipeline that wraps directories in a `<details>` element to make them collapsible.

Each filename is passed to a utility based on the Seti UI VS Code icon system to get an icon for the detected file type. This is injected as an inline SVG.

This pipeline also separates the comments and interprets the bold filename as highlighted.

### CODE REVIEW

**Comment by sarah11918** [comment]
Super excited for this @delucis! I tested all the docs pages, at various screen sizes and on my phone, and it all looks AMAZINGLY good to me!  🥳

**Review by natemoo-re — APPROVED** [review]
This is so cool!

**Review by Yan-Thomas — APPROVED** [review]
Tested with NVDA, it's perfect! Great work @delucis 🙌

File: src/pages/en/core-concepts/project-structure.md [hunk 1]
```diff
@@ -23,30 +25,29 @@ Astro leverages an opinionated folder layout for your project. Every Astro proje
...
+- src/
+  - components/
+    - Header.astro
+    - Button.jsx
+  - layouts/
+    - PostLayout.astro
+  - pages/
+    - posts/
+      - post1.md
+      - post2.mdx
```
Comment (delucis) [inline src/pages/en/core-concepts/project-structure.md:42]: Change from when I was testing snuck in. Should change back but also making a note for tomorrow to add a file association/icon for MDX

```suggestion
      - post2.md
```

**Pull request merged**